File: problems/1547-destination-city/solution.go
func destCity(paths [][]string) string {
	start := make(map[string]int)
	destination := make(map[string]int)

	for _, path := range paths {
		start[path[0]] = 1
		destination[path[1]] = 1
		
		if destination[path[0]] == 1 {
			delete(destination, path[0])
			delete(start, path[0])
		}
		if start[path[1]] == 1 {
			delete(start, path[1])
			delete(destination, path[1])
		}
	}
	
	for s, _ := range destination {
		return s
	}
	return ""
}

File: problems/1547-destination-city/solution.java
class Solution {
    public String destCity(List<List<String>> paths) {
        String isEndDestination = "";
        for (int i = 0; i < paths.size(); i++) {
            isEndDestination = paths.get(i).get(1);
            boolean end = true;
            for (int j = 0; j < paths.size(); j++) {
                if (paths.get(j).get(0).equals(isEndDestination)){
                    end = false;
                    break;
                }
            }
            if (end){
                return isEndDestination;
            }
        }
        return isEndDestination;
    }
}